common/io/Serial.cpp

```diff
@@ -31,6 +46,59 @@ namespace io {
 using std::vector;
 using std::string;
 
+namespace {
+
+string GetLockFile(const string &path) {
+  const string base_name = ola::file::FilenameFromPath(path);
```

Comment:
I think we had this discussion before, but this potentially falls down if some files aren't in /dev.